src/cdk/explorer/nodes/appNode.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { AWSTreeNodeBase } from '../../../shared/treeview/nodes/awsTreeNodeBase'
+import { PlaceholderNode } from '../../../shared/treeview/nodes/placeholderNode'
+import { cdk } from '../../globals'
+import { CdkAppLocation, getApp } from '../cdkProject'
+import { includeConstructInTree } from '../tree/treeInspector'
+import { ConstructNode } from './constructNode'
+
+/**
+ * Represents a CDK App
+ * Existence of apps is determined by the presence of `cdk.json` in a workspace folder
+ */
+export class AppNode extends AWSTreeNodeBase {
+    get tooltip(): string {
+        return this.app.cdkJsonPath
+    }
+
+    get id(): string {
+        return this.app.treePath
```

Comment:
Is it possible for two different files to produce the same treePath? If so, you'll need to do something like append the json path into here to guarantee id values are unique

Reply:
yes they can, I'd worked this change into my subsequent commit. let me know if you'd like me to fold it into this one

Reply:
Subsequent PR is ok (please add a TODO into this one)

Reply:
done